### PR TITLE
Implement is_tuple_like

### LINKED ISSUE #1667: Implement is_tuple_like
At a moment  in  `tuple_util.hpp` the `tuple_like` concept is expressed only in comments. It is possible to have it as a type trait (C++17) and as a concept (C++20).
I would propose to place this trait in `gridtools::tuple_util` namespace and expose it to `gridtools` by `using`.
As for concepts, my proposal is to place it in `gridtools::concepts`.
The predicate should check:
  * `gridtools::traits::to_types<T>` exists and it is a type list (`meta::is_list`)
  * `gridtools::traits::from_types<T>` exists and if we "apply" it to `to_types<T>` we get `T` 
  *  element wise constructor exists.
  *  if all  types in `to_types<T>` are move constructible `T` is also move constructible
  * same for copy ctors
  * `gridtools::traits::getter<T>`  exsists and for each index from `0`  to `meta::length<to_types<T>>::value` is should be checked that `getter<T>::template get<I>`  returns the right type.

### PR DESCRIPTION
**Pull request by anstaf**
resolves #1667

### CODE REVIEW

**Comment by anstaf** [comment]
launch

File: include/gridtools/common/tuple_util.hpp [hunk 1]
```diff
@@ -411,8 +414,58 @@ namespace gridtools {
...
+            > {};
+
+            template <class T,
+                class Types = traits::to_types<T>,
+                class FromTypes = traits::from_types<T>,
+                class Getter = traits::getter<T>>
+            struct is_tuple_like : bool_constant<
+                meta::is_list<Types>::value &&
+                meta::all_of<meta::not_<std::is_void>::apply, Types>::value &&
+                std::is_same<meta::rename<FromTypes::template apply, Types>, T>::value &&
```
Comment (havogt) [inline include/gridtools/common/tuple_util.hpp:441]: what's this about?

Comment (havogt) [reply include/gridtools/common/tuple_util.hpp:441]: Can you add to the description at the top of the file of `tuple_to_types` and `tuple_from_types` an example for a non trivial case?

**Review by havogt — CHANGES_REQUESTED** [review]
Looks good, just a small request to add additional examples for the behavior of tuple_like

**Comment by havogt** [comment]
launch jenkins

File: include/gridtools/common/tuple_util.hpp (hunk 1 above)
Comment (anstaf) [reply include/gridtools/common/tuple_util.hpp:446]: I added more comments.
I am not sure what non trivial case you have in mind.
Can you write github suggestion?

Comment (havogt) [reply include/gridtools/common/tuple_util.hpp:446]: `to_types<T>` for all the normal cases will probably return just itself (`T`). My question was, what's the example where it does not. But in the meantime I answered myself: `array`.

**Review by havogt — APPROVED** [review]

**Comment by havogt** [comment]
tests ok.

**Pull request merged**